### ISSUE TITLE
Deduplicate test suite with something generic.

### ISSUE THREAD

**Issue by pipermerriam** [ethereum/async-service#19]
### What is wrong

Currently, the test suite is basically duplicated between the `tests-asyncio` and `tests-trio` directories.  

### How can it be fixed

Some sort of framework agnostic test suite that can be run within either `asyncio` or `trio`.

Probably depends on #18

**Comment by staccDOTsol** [comment]
Yes, I agree that having a duplicated test suite between the `tests-asyncio` and `tests-trio` directories is not ideal. It would be more efficient to have a framework-agnostic test suite that can be run within either `asyncio` or `trio`.

One possible solution could be to create a separate directory, let's say `tests-common`, where you can place the generic test cases that are applicable to both `asyncio` and `trio`. This way, you can avoid duplicating the test suite code.

To achieve this, you can refactor the existing test cases by extracting the common parts into separate modules or functions within the `tests-common` directory. Then, you can import and utilize these common test cases in both the `tests-asyncio` and `tests-trio` directories.

Additionally, it seems like this improvement might depend on issue #18. It would be helpful to address that issue first to ensure a smooth integration of the framework-agnostic test suite.

Let me know if you need any further assistance or clarification!
